File: vertx-lang-php-jphp/src/main/php/io/vertx/php/core/datagram/DatagramSocket.php
<?php 
namespace io\vertx\php\core\datagram;
use io\vertx\php\core\buffer\Buffer;
use io\vertx\php\core\net\SocketAddress;
use io\vertx\php\core\streams\WriteStream;
class DatagramSocket
{
    
    private function __construct()
    {

    }
    /**
     * (string multicastAddress, string sourceToBlock, callable handler)
     * @return DatagramSocket
     * (string multicastAddress, string networkInterface, string sourceToBlock, callable handler)
     * @return DatagramSocket
     * methodSize : 2
     */
    public function blockMulticastGroup(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * (callable handler)
     * @return Void
     * ()
     * @return Void
     * methodSize : 2
     */
    public function close(/*1*/$arg0)
    {

    }
    /**
     * (callable endHandler)
     * @return DatagramSocket
     * methodSize : 1
     */
    public function endHandler(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return DatagramSocket
     * methodSize : 1
     */
    public function exceptionHandler(/*1*/$arg0)
    {

    }
    /**
     * (callable handler)
     * @return DatagramSocket
     * methodSize : 1
     */
    public function handler(/*1*/$arg0)
    {

    }
    /**
     * ()
     * @return bool
     * methodSize : 1
     */
    public function isMetricsEnabled(/*0*/)
    {

    }
    /**
     * (int port, string host, callable handler)
     * @return DatagramSocket
     * methodSize : 1
     */
    public function listen(/*3*/$arg0, $arg1, $arg2)
    {

    }
    /**
     * (string multicastAddress, callable handler)
     * @return DatagramSocket
     * (string multicastAddress, string networkInterface, string source, callable handler)
     * @return DatagramSocket
     * methodSize : 2
     */
    public function listenMulticastGroup(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
    /**
     * ()
     * @return SocketAddress
     * methodSize : 1
     */
    public function localAddress(/*0*/)
    {

    }
    /**
     * ()
     * @return DatagramSocket
     * methodSize : 1
     */
    public function pause(/*0*/)
    {

    }
    /**
     * ()
     * @return DatagramSocket
     * methodSize : 1
     */
    public function resume(/*0*/)
    {

    }
    /**
     * (Buffer packet, int port, string host, callable handler)
     * @return DatagramSocket
     * (string str, int port, string host, callable handler)
     * @return DatagramSocket
     * (string str, string enc, int port, string host, callable handler)
     * @return DatagramSocket
     * methodSize : 3
     */
    public function send(/*5*/$arg0, $arg1, $arg2, $arg3, $arg4)
    {

    }
    /**
     * (int port, string host)
     * @return WriteStream
     * methodSize : 1
     */
    public function sender(/*2*/$arg0, $arg1)
    {

    }
    /**
     * (string multicastAddress, callable handler)
     * @return DatagramSocket
     * (string multicastAddress, string networkInterface, string source, callable handler)
     * @return DatagramSocket
     * methodSize : 2
     */
    public function unlistenMulticastGroup(/*4*/$arg0, $arg1, $arg2, $arg3)
    {

    }
}

File: vertx-lang-php-jphp/src/main/test/io/vertx/php/core/datagram/DatagramSocket.java
package io.vertx.php.core.datagram;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Namespace;
import php.runtime.annotation.Reflection.Signature;
import io.vertx.lang.jphp.IWrapper;
import io.vertx.php.core.buffer.Buffer;
import io.vertx.php.core.datagram.DatagramPacket;
import io.vertx.php.core.metrics.Measured;
import io.vertx.php.core.streams.WriteStream;
import io.vertx.php.core.streams.ReadStream;
import io.vertx.php.core.net.SocketAddress;
@Name("DatagramSocket")
@Namespace("io\\vertx\\php\\core\\datagram")
public interface DatagramSocket<S extends io.vertx.core.datagram.DatagramSocket> extends IWrapper<S>{
    
    @Signature
    default Memory isMetricsEnabled(Environment __env__, Memory... args) {
        return Utils.convReturnUnknow(__env__, this.getWrappedObject().isMetricsEnabled()
    }

    @Signature
    default Memory send(Environment __env__, Memory... args) {
        this.getWrappedObject().send(packet, port, host, handler)
    }

    @Signature
    default Memory sender(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, WriteStreamImpl::new, this.getWrappedObject().sender(port, host)
    }
    
    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }

    @Signature
    default void close(Environment __env__, Memory... args) {
        this.getWrappedObject().close()
    }
    
    @Signature
    default Memory localAddress(Environment __env__, Memory... args) {
        return Utils.convReturnVertxGen(__env__, SocketAddressImpl::new, this.getWrappedObject().localAddress()
    }

    @Signature
    default Memory listenMulticastGroup(Environment __env__, Memory... args) {
        this.getWrappedObject().listenMulticastGroup(multicastAddress, handler)
    }

    @Signature
    default Memory unlistenMulticastGroup(Environment __env__, Memory... args) {
        this.getWrappedObject().unlistenMulticastGroup(multicastAddress, handler)
    }

    @Signature
    default Memory blockMulticastGroup(Environment __env__, Memory... args) {
        this.getWrappedObject().blockMulticastGroup(multicastAddress, sourceToBlock, handler)
    }

    @Signature
    default Memory listen(Environment __env__, Memory... args) {
        this.getWrappedObject().listen(port, host, handler)
    }
    
    @Signature
    default Memory pause(Environment __env__, Memory... args) {
        this.getWrappedObject().pause()
    }
    
    @Signature
    default Memory resume(Environment __env__, Memory... args) {
        this.getWrappedObject().resume()
    }

    @Signature
    default Memory endHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().endHandler(endHandler)
    }

    @Signature
    default Memory handler(Environment __env__, Memory... args) {
        this.getWrappedObject().handler(handler)
    }

    @Signature
    default Memory exceptionHandler(Environment __env__, Memory... args) {
        this.getWrappedObject().exceptionHandler(handler)
    }

}
